tools/build_file_generator/ThirdPartyDepsTool.java

```diff
@@ -18,39 +18,61 @@
 
 import com.google.startupos.common.CommonModule;
 import com.google.startupos.common.FileUtils;
-import com.google.startupos.common.flags.Flag;
-import com.google.startupos.common.flags.FlagDesc;
-import com.google.startupos.common.flags.Flags;
+import com.google.startupos.tools.buildfilegenerator.Protos.ThirdPartyDeps;
 import dagger.Component;
 
 import javax.inject.Singleton;
 import java.io.IOException;
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.HashMap;
+import java.util.Map;
 
-import com.google.startupos.tools.buildfilegenerator.Protos.ThirdPartyDeps;
-
-/* A tool for generating prototxt file to know which classes each third_party dependency includes
- *
- * Note! Before using this tool, it is necessary to run `./build.sh`
- * to have all third_party deps in `bazel-startup-os/external` folder
+/* A tool for generating the zip-archive with prototxt file inside
+ * to know which classes each third_party dependency includes.
+ * Path to the zip-archive: "//tools/build_file_generator/third_party_deps.zip"
  *
  * Usage:
- * bazel run //tools/build_file_generator:third_party_deps_tool -- --file_path </path/to/file>
+ * bazel run //tools/build_file_generator:third_party_deps_tool
  */
 public class ThirdPartyDepsTool {
-  @FlagDesc(name = "file_path", description = "Absolute path to file")
-  private static Flag<String> filePath = Flag.create("");
-
   public static void main(String[] args) throws IOException {
-    Flags.parseCurrentPackage(args);
-    checkInputFlags();
-
     ThirdPartyDepsToolComponent component =
         DaggerThirdPartyDepsTool_ThirdPartyDepsToolComponent.create();
 
-    ThirdPartyDeps thirdPartyDeps = component.getThirdPartyDepsAnalyzer().getThirdPartyDeps();
+    createZip(component.getFileUtils(), component.getThirdPartyDepsAnalyzer().getThirdPartyDeps());
+  }
 
-    component.getFileUtils().writePrototxtUnchecked(thirdPartyDeps, filePath.get());
+  private static void createZip(FileUtils fileUtils, ThirdPartyDeps thirdPartyDeps)
+      throws IOException {
+    final String prototxtFile = "third_party_deps.prototxt";
+    String absBuildFileGeneratorDirPath =
+        fileUtils.joinPaths(fileUtils.getCurrentWorkingDirectory(), "tools/build_file_generator");
+
+    Map<String, String> env = new HashMap<>();
+    env.put("create", "true");
+
+    URI uri =
+        URI.create(
+            String.format(
+                "jar:file:%s",
+                fileUtils.joinPaths(absBuildFileGeneratorDirPath, "third_party_deps.zip")));
+    try (FileSystem zipfs = FileSystems.newFileSystem(uri, env)) {
```

Comment:
How about instead of writing a file, copying and then deleting, you add a new method to `fileUtils` called `writePrototxtToZip` that writes directly to the Zip file?

Reply:
Done.